python/tests/io/test_base.py

```diff
@@ -0,0 +1,188 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import tempfile
+
+import pytest
+
+from iceberg.io.base import FileIO, InputFile, OutputFile
+
+
+class LocalInputFile(InputFile):
+    """An InputFile implementation for local files (for test use only)"""
+
+    def __init__(self, location: str):
+        if not location.startswith("file://"):
```

Comment:
This could be `file:///` or `file:/`. The first has an authority section (after `//`) but it is empty. The second variation leaves out authority and just has a path. Either way, the path is a full path starting from `/`. Also, one case that is _not_ allowed is a URI like `file://one/two/three/a.parquet` because the authority is `one` and no authority is allowed for local FS URIs.

Reply:
I switched to using `urllib.parse.urlparse` which is commonly used in other packages. This returns a `ParseResult` where you can check `ParseResult.scheme`, `ParseResult.path`, etc. and I set that to a property called `parsed_result`.

In addition to checking that the scheme is `file`, I also added a check that there's no `PraseResult.netloc` for a `LocalInputFile` or `LocalOutputFile`, which is the authority section.

relevant commit: [fcb7dc4](https://github.com/apache/iceberg/pull/3691/commits/fcb7dc48f879d9ac3785c0ec274264a44e32ff3a)